touchforms/formplayer/static/formplayer/script/formnav-all.js

```diff
@@ -9,6 +10,20 @@ function xformAjaxAdapter (formSpec, sessionData, savedInstance, ajaxfunc, submi
   this.render_context = render_context;
   this.answerCallback = answerCallback;
 
+  $.subscribe('formplayer.submit-form', function(e, form) {
```

Comment:
this is one of the bigger paradigm shifts. before there was barely any separation of the model, the controller or the view. now we can interface the controller (adapter) via a publish and subscribe interface. this makes it way easier to test. we don't have to mock the adapter and only check to make sure we publish the correct events